### PR TITLE
Update shiny gadget with dynamic UI to cater for all 4 project types

### PR DESCRIPTION
**Pull request by jonmcalder**
See #12 and #15

### CODE REVIEW

**Comment by stephlocke** [comment]
FYI this may error it's check as coveralls.io is down right now

File: R/shinyGadget.R [hunk 1]
```diff
@@ -1,29 +1,98 @@
...
+  server <- function(input, output, session) {
 
-  shiny::observeEvent(input$done, {
-    createBasicProject(name = input$name, readme = input$readme,
-                       git = input$git, travis = input$travis,
-                       packrat = input$packrat)
-    shiny::stopApp()
-  })
-}
+    output$ui <- renderUI({
```
Comment (stephlocke) [inline R/shinyGadget.R]: this needs a `shiny::` prefix
> projectGadget : server: no visible global function definition for ‘renderUI’

File: R/shinyGadget.R [hunk 2]
```diff
@@ -1,29 +1,98 @@
...
+
+      switch(input$projectType,
+
+        "Basic"     = shiny::div(),
+        "Analysis"  = shiny::inputPanel(
+                        shiny::selectInput("dirs", label = "Directories",
```
Comment (stephlocke) [inline R/shinyGadget.R]: @jonmcalder Do you think people will object to the defaults and/or prefer their own such that this should be some sort of free text field?

Comment (jonmcalder) [inline R/shinyGadget.R]: I had the same thought but was lazy so just did it the "neat" way to start with.

In general (not just in this instance) I think it's probably a good idea to try and make things as flexible as possible. But one also has to balance that with efficiency - e.g. we could allow custom text inputs here, but I'd argue that typing directory names into the gadget isn't really much more efficient than typing them out in the function call directly or just creating them manually. So on a case by case basis, that flexibility isn't really worth much.

If more and/or different directory names are needed on a repeated basis then I'd say this is another potentially good candidate for the yaml template we discussed previously. Then this directory list could be populated from there which should give the best of both worlds (flexibility for repeated cases + better efficiency)

**Pull request merged**